core/windows_and_tabs/window_management.talon

```diff
@@ -17,5 +17,43 @@ snap last [screen]: user.move_window_previous_screen()
 snap screen <number>: user.move_window_to_screen(number)
 snap <user.running_applications> <user.window_snap_position>:
     user.snap_app(running_applications, window_snap_position)
+snap left <user.running_applications> <user.running_applications>:
+    user.switcher_focus(running_applications_1)
+    user.snap_window_to_position("left")
```

Comment:
`user.snap_app` combines these actions, which will reduce the code duplication here.

In theory, it would also allow us to make the rearrangement faster and more elegant -- remember that we can move/resize windows without having to focus the applications they belong to. Although, I imagine you always focus to make sure that they aren't occluded by other applications, and that the other application is second in your application switcher LRU.

Reply:
snap_app doesn't take a string like "right" but a more complex object-I'll need to make a new version of that method to do this, but that's ok I think.

Reply:
You would just need to index into `_snap_positions`. ` <user.window_snap_position>` effectively does this for you if used, but it should be fine to have an action that also takes the string name (seems useful)

Reply:
The only problem with using despot can form as an identifier is that if you use it in a script somewhere and then changed the spoken form your script is now broken. If you want to have an action that takes a string we should probably separate the spoken form from the identifier.

Reply:
Sorry, I didn't quite follow the suggestion-what do you think we should do, @AndreasArvidsson ?

Reply:
Just that the list of available snap positions should have two different strings. One that is a spoken form and one that is an identifier. If I changed a spoken form from `left` to `lefter` the code using `left` is not valid anymore.

Reply:
Makes sense. Is this typically done with a talon list?

Reply:
Not typically, but that's definitely one good implementation. Look at four matters for example. 

Here the keys are the identifier. These you can use from your own scripts and they will work even if you changed the spoken form.
https://github.com/talonhub/community/blob/7619d2c369c189667f1ba5f0089b843517e6a507/core/text/formatters.py#L133-L170

Here is the map where the keys is the spoken form that maps to the identifiers. This could have been a Talon list file
https://github.com/talonhub/community/blob/7619d2c369c189667f1ba5f0089b843517e6a507/core/text/formatters.py#L173-L189

Reply:
Thanks that makes sense. I changed this in favor of using the internal apis which I think addresses this concern.